From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct se_cmd
Patch-mainline: never, kabi

In 3.12.43, commit 15a557ae69ea1e86aa8db5c9a48f7406d9235b72 (target:
Fix COMPARE_AND_WRITE with SG_TO_MEM_NOALLOC handling), upstream
commit c8e639852ad720499912acedfd6b072325fd2807 and in 3.12.56, commit
efc98f01dc98e7a4169df578f109d36608c60b28 (target: Fix race for
SCF_COMPARE_AND_WRITE_POST checking), upstream commit
057085e522f8bf94c2e691a5b76880f68060f8ba changed prototype of one hook
in struct se_cmd. It indeed changed the layout of that structure.

Since it did not change the size of that structure and is only
relevant to target_core_sbc, we can just protect the change the usual
way, by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/target/target_core_base.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/target/target_core_base.h
+++ b/include/target/target_core_base.h
@@ -457,7 +457,11 @@ struct se_cmd {
 	sense_reason_t		(*execute_cmd)(struct se_cmd *);
 	sense_reason_t		(*execute_rw)(struct se_cmd *, struct scatterlist *,
 					      u32, enum dma_data_direction);
+#ifdef __GENKSYMS__
+	sense_reason_t (*transport_complete_callback)(struct se_cmd *);
+#else
 	sense_reason_t (*transport_complete_callback)(struct se_cmd *, bool, int *);
+#endif
 
 	unsigned char		*t_task_cdb;
 	unsigned char		__t_task_cdb[TCM_MAX_COMMAND_SIZE];
